.github/workflows/trunk.yml

```diff
@@ -55,6 +55,31 @@ jobs:
         # Build and test executorch
         PYTHON_EXECUTABLE=python ${CONDA_RUN} bash .ci/scripts/test_model.sh "${MODEL_NAME}" "${BUILD_TOOL}" "${BACKEND}"
 
+  test-models-macos-coreml:
+    name: test-models-macos-coreml
+    uses: pytorch/test-infra/.github/workflows/macos_job.yml@main
+    strategy:
+      matrix: 
+        model: [dl3, edsr, efficient_sam, emformer_join, emformer_transcribe, ic3, mobilebert, mv3, resnet50, vit, w2l]
+        backend: [coreml]
```

Comment:
How is this different from `test-apple-model` besides number of models? Ah no mps. But so why do we have coreml there as well?

Reply:
Yeah good point. 

I removed coreml stuff from test-apple-model and left the mps only. And renamed test-models-macos-mps to be consistent